File: ideal-job-scheduler/ideal-job-scheduler-core/src/main/kotlin/cn/sh/ideal/job/scheduler/core/admin/service/JobInstanceService.java
package cn.sh.ideal.job.scheduler.core.admin.service;

import cn.sh.ideal.job.scheduler.core.admin.entity.JobInstance;
import cn.sh.ideal.job.scheduler.core.admin.repository.JobInstanceRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import java.time.LocalDateTime;

/**
 * @author 宋志宗
 * @date 2020/9/2
 */
@SuppressWarnings("UnusedReturnValue")
@Service
public class JobInstanceService {
    private static final Logger log = LoggerFactory.getLogger(JobInstanceService.class);
    private static final int MAX_RESULT_LENGTH = 10000;
    private final JobInstanceRepository jobInstanceRepository;

    public JobInstanceService(JobInstanceRepository jobInstanceRepository) {
        this.jobInstanceRepository = jobInstanceRepository;
    }

    @Nonnull
    public JobInstance saveInstance(@Nonnull JobInstance instance) {
        String result = instance.getResult();
        if (result.length() > MAX_RESULT_LENGTH) {
            instance.setResult(result.substring(0, MAX_RESULT_LENGTH - 3) + "...");
        }
        return jobInstanceRepository.save(instance);
    }

    @Nullable
    public JobInstance getJobInstance(long instanceId) {
        return jobInstanceRepository.findById(instanceId).orElse(null);
    }

    public void updateDispatchInfo(@Nonnull JobInstance instance) {
        jobInstanceRepository.updateDispatchInfo(instance);
    }

    public int updateWhenTriggerCallback(@Nonnull JobInstance instance) {
        return jobInstanceRepository.updateWhenTriggerCallback(instance);
    }

    public int deleteAllByCreatedTimeLessThan(LocalDateTime time) {
        return jobInstanceRepository.deleteAllByCreatedTimeLessThan(time);
    }

    public void flush() {
        jobInstanceRepository.flush();
    }
}

File: ideal-job-scheduler/ideal-job-scheduler-core/src/main/kotlin/cn/sh/ideal/job/scheduler/core/admin/service/JobInstanceService.kt
package cn.sh.ideal.job.scheduler.core.admin.service

import cn.sh.ideal.job.scheduler.core.admin.entity.JobInstance
import cn.sh.ideal.job.scheduler.core.admin.repository.JobInstanceRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import java.time.LocalDateTime

/**
 * @author 宋志宗
 * @date 2020/8/23
 */
@Service
class JobInstanceService(private val jobInstanceRepository: JobInstanceRepository) {
  private val maxResultLength = 10000


  fun saveInstance(instance: JobInstance): JobInstance {
    val result = instance.result
    if (result.length > maxResultLength) {
      instance.result = instance.result.substring(0, maxResultLength - 3).plus("...")
    }
    return jobInstanceRepository.save(instance)
  }

  fun getJobInstance(jobInstanceId: Long): JobInstance? {
    return jobInstanceRepository.findByIdOrNull(jobInstanceId)
  }


  fun updateDispatchInfo(instance: JobInstance) {
    jobInstanceRepository.updateDispatchInfo(instance)
  }

  fun updateWhenTriggerCallback(instance: JobInstance): Int {
    return jobInstanceRepository.updateWhenTriggerCallback(instance)
  }

  fun deleteAllByCreatedTimeLessThan(time: LocalDateTime): Int {
    return jobInstanceRepository.deleteAllByCreatedTimeLessThan(time)
  }

  fun flush() {
    jobInstanceRepository.flush()
  }
}